### ISSUE TITLE
Support for different number of input channels

### ISSUE THREAD

**Issue by weiji14** [developer0hye/YOLOX-Backbone#1]
Hi there, thanks for making this lightweight package of YOLOX!

I was wondering if the `create_model` function could be modified to support more inputs channels? At the moment it is hardcoded to support only 3 channels (RGB), but I have a project that will have 4 (or more) channel inputs (e.g. RGBA). Scanning through the code, I think it would require modifying the line below (and tracing the 3 up a few classes back to `create_model`)

https://github.com/developer0hye/YOLOX-Backbone/blob/681752763ca7dd4db4ecc18693fec1c6cd2e34b2/yolox_backbone/models/darknet.py#L115

 Happy to submit a PR if it helps, I just wanted to check if this repo is meant to follow the original YOLOX implementation, or if there is room to expand some more :grin:

**Comment by developer0hye** [comment]
@weiji14 

Hi, weiji

Thanks for your idea!

I have one question. If we set the input channel of the network, how should we deal with pre-trained weight?

Should we just load all the weights except the first layer in the stem block?

**Comment by weiji14** [comment]
Personally I don't use the pretrained weights, but yeah, I imagine loading all the weights except for layer 1 would be the way to go. Though not exactly sure how to implement that in practice...

**Comment by developer0hye** [comment]
@weiji14 

I updated the code for your case.

Now you can create the model without the constraint that the number of input channel is 3.
```python
model = yolox_backbone.create_model(model_name=model_name, 
                                    pretrained=False, 
                                    input_tensor_channels=4,
                                    out_features=["P3", "P4", "P5"]
                                    )
```

Please check it!

Tell me if you find anything

**Comment by weiji14** [comment]
Wow that was super fast! I've tested v0.0.1.8 and the `input_tensor_channels` parameter works like a charm, thank you so much! Will close this issue now, but feel free to reopen if you want to support `input_tensor_channels !=3` and `pretrained=True` :nerd_face:

**Comment by developer0hye** [comment]
@weiji14 I am interested in your work. Do you have a plan to open your research you have been studying with this project?

**Comment by weiji14** [comment]
Hi @developer0hye, yes I'll definitely be open sourcing the work :grin:. The project is just starting up, but you can take a sneak peek at the code [here](https://dagshub.com/weiji14/h2oval/src/fec4ae18873e20954510728fd81fc307478e0e1b/h2oval/h2ovalnet.py#L76-L81). I'm working with optical satellite images from [Sentinel 2](https://en.wikipedia.org/wiki/Sentinel-2), which has different channels (or 'bands' as we like to call it). Will probably migrate the code over to GitHub at some point, just wanted to try out some `dvc` features on DAGsHub for a change.

**Comment by developer0hye** [comment]
@weiji14 
Cool work! I hope you successfully to finish your project! Is DVC good for managing project? Now, I and coworkers have a hard time of managing dataset... We are managing dataset without any managing tool like DVC. The bigger the scale of dataset is, the harder we manage it...

**Comment by weiji14** [comment]
Thanks :smiley: DVC is a pretty good tool, though I've only used it for small projects so far. Have a look at their docs on https://dvc.org, or this blog post https://mmeendez8.github.io/2021/08/30/mmdet-dvc-tutorial.html. The `dvc` syntax is similar to `git` (e.g. `dvc add`, `dvc push`, etc), so it helps lower the barrier. I found that `dvc` works is a great tool for data versioning, but there are some new features that handle experiment metric tracking as well, so I think it's definitely worth trying out!
